webui/react/src/services/utils.ts

```diff
@@ -59,7 +68,11 @@ export const readStream = async <T = unknown>(
       if (isCancelled) return;
       try {
         const ndjson = JSON.parse(line);
-        onEvent?.(ndjson.result);
+        if (ndjson.error) {
+          onError?.(ndjson.error);
+        } else {
+          onEvent?.(ndjson.result);
+        }
```

Comment:
this handles errors that happen mid-stream. 

kinda hacky, but if you look at the network tab, you will see `200 OK` when `grpcutil` returns an error in the middle of streaming, so I don't think there's any significantly better option.